File: ProgrammingChallenges#Book/Chapter 1/The Trip/solutions/TheTrip.py
from sys import stdin

# Main program
def main():
    expenses = [0]*1000

    for line in stdin:
        n = int(line)

        if (n == 0):
            break;

        total, toExchangePos, toExchangeNeg = (0,)*3

        for i in range(n):
            line = stdin.readline()
            expenses[i] = float(line)
            total += expenses[i]

        # Get average
        average = total / n

        for i in range(n):
            dif = expenses[i] - average
            # Set two digits accuracy
            dif = float((int(dif * 100.0)) / 100.0)

            if (dif > 0):
                toExchangePos += dif
            else:
                toExchangeNeg += dif

        if (-toExchangeNeg > toExchangePos):
            minToExchange = -toExchangeNeg
        else:
            minToExchange = toExchangePos

        print("$%.2f" % minToExchange)

main()

File: ProgrammingChallenges#Book/Chapter 1/03.TheTrip/solutions/TheTrip.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 *
 * @author xergioalex
 */
public class TheTrip {
    static double[] expenses = new double[1000];

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        String inputLine;

        while((inputLine = reader.readLine()) != null) {
            int n = Integer.parseInt(inputLine);

            if (n == 0) {
                break;
            }

            double total = 0, average = 0, toExchangePos = 0, toExchangeNeg = 0;

            for (int i = 0; i < n; i++) {
                inputLine = reader.readLine();
                expenses[i] = Double.parseDouble(inputLine);
                total += expenses[i];
            }

            /* Get average */
            average = total / (double) n;

            for (int i = 0; i < n; i++) {
                double dif = expenses[i] - average;
                /* Set two digits accuracy */
                dif = (double) (((long) (dif * 100.0)) / 100.0 );

                if (dif > 0) {
                    toExchangePos += dif;
                } else {
                    toExchangeNeg += dif;
                }
            }

            double minToExchange = (-toExchangeNeg > toExchangePos)? -toExchangeNeg: toExchangePos;
            System.out.format("$%.2f\n", minToExchange);
        }
    }
}